test/fit_tests/tests/rackhd20/test_rackhd20_api_tagmanagement.py

```diff
@@ -0,0 +1,401 @@
+import fit_common
+import test_api_utils
+
+# Local methods
+MON_NODES = fit_common.node_select()
+MON_API_VERSION = 2.0
+
+#clear the test environment
+def tear_down():
+    str_taglist = get_tag_list()
+    taglist = json.loads(str_taglist)
+    for tagitem in taglist:
+        tagname = tagitem['name']
+        delete_response = delete_tag(tagname)
+        if delete_response not in range(200,205):
+            print "Fail to clear up the tag-"+ tagname
+            return 1
+    nodes_list = get_node_id_list()
+    for node in nodes_list:
+        tags = get_tag_list_on_node(node)
+        tag_list = json.loads(tags)
+        for tag in tag_list:
+            print "{} has tags: {}".format(node,tag)
+            response = delete_tag_in_node(node,tag)
+            if response  not in range(200,205):
+                print "Fail to delete the tag:" + tag
+                return 2
+    print "clear the environment finished!"
+    return 0
+
+def get_node_list_by_sku(skuname):
+    response = fit_common.rackhdapi("/api/{}/skus".format(MON_API_VERSION))
+    skus = json.loads(response['text'])
+    idfound = []
+    for sku in skus:
+        if skuname == sku['name']:
+            sku_id = sku['id']
+    response = fit_common.rackhdapi('/api/{}/nodes'.format(MON_API_VERSION))
+    nodes = json.loads(response['text'])
+    for node in nodes:
+        if MON_API_VERSION == 1.1:
+            if node.has_key('sku') and node['sku'] == sku_id:
+                idfound.append(node['id'])
+        if MON_API_VERSION == 2.0:
+            if node.has_key('sku') and node['sku'].split('/')[-1] == sku_id:
+                idfound.append(node['id'])
+    return idfound
+    
+def delete_tag(tag_name):
+    url_name = urllib.quote(tag_name,"")
+    mon_url = "/api/{}/tags/{}".format(MON_API_VERSION,url_name)
+    response_tag = fit_common.rackhdapi(mon_url,action='delete')
+    return response_tag["status"]
+
+def create_tag(json_tag_content):
+    mon_url = '/api/{}/tags'.format(MON_API_VERSION)
+    response_create_tag = fit_common.rackhdapi(mon_url,action='post',payload=json_tag_content)
+    return response_create_tag["status"]
+
+def get_tag(tag_name):
+    url_name = urllib.quote(tag_name,"")
+    mon_url = '/api/{}/tags/{}'.format(MON_API_VERSION,url_name)
+    response_tag = fit_common.rackhdapi(mon_url)
+    return response_tag['text']
+
+def get_node_by_tag(tag_name):
+    url_name = urllib.quote(tag_name,"")
+    mon_url = '/api/{}/tags/{}/nodes'.format(MON_API_VERSION,url_name)
+    response_tag = fit_common.rackhdapi(mon_url)
+    if response_tag['status'] in range(200,205):
+        return response_tag['text']
+    else:
+        return response_tag['status']
+
+def get_tag_list():
+    mon_url = '/api/{}/tags'.format(MON_API_VERSION)
+    response_tag = fit_common.rackhdapi(mon_url)
+    return response_tag['text']
+
+def get_tag_list_on_node(nodeid):
```

Comment:
It seems that the swagger client python library is not in fit_test folder. If only using FIT to run tests, then the library cannot be imported. So I did not use the library.